### PR TITLE
Extend $ubiModal options

### PR DESCRIPTION
**Pull request by mdottavio**
#### What's this PR do?
Extend the options provided to the component `$ubiModal`; 
By default, the component will still make the `modal` with the default template but you could also send the `templateUrl` param to use a custom layout or `template` to build your own `html`.
A list of options the `$ubiModal` component accepts [can be found here](http://angular-ui.github.io/bootstrap/versioned-docs/1.3.0/#/modal).

**This is a semver:feature**

Clean review using [`w=1`](https://github.com/Olapic/angular-popup-boxes/pull/8/files?w=1)

#### Where should the reviewer start?
You can review the code and the test the implementation on `lemuramaAdmin` (the modal's behavior shouldn't change) or review the related PR ;)

### CODE REVIEW

File: angular-popup-boxes.js [hunk 1]
```diff
@@ -8,137 +8,152 @@ angular.module('angularPopupBoxes', ['ui.bootstrap'])
...
-                modalEl.width(options.width);
-                modalEl.find(".angular-notification-btn-ok").click(function() {
-                    modal.close();
+          var modal = $modal.open(modalOpts);
+
+          setTimeout(function () {
+              modalEl = $('#' + modalOpts.modalId).parent().parent();
+              modalEl.parent().css('display', '');
+              modalEl.width(options.width);
+              debugger;
```
Comment (mdottavio) [inline angular-popup-boxes.js]: :shame:

**Comment by emiprandi** [comment]
Code looks good, but there are many conventions issues: extra spaces, comma at the end of object properties, a mix of 4 and 2 spaces indentation that should be fixed.

**Comment by mdottavio** [comment]
@emiprandi true; should I rewrite the whole file?

File: angular-popup-boxes.js [hunk 2]
```diff
@@ -8,137 +8,151 @@ angular.module('angularPopupBoxes', ['ui.bootstrap'])
...
+            okText: '',
+            cancelText: '',
+            okClass: '',
+            modalId: 'modal-' + Math.floor(Math.random() * 9999),
+          };
+
+          angular.extend(modalOpts, content);
+          var modalEl;
+
+          if (modalOpts.templateUrl === undefined) {
```
Comment (emiprandi) [inline angular-popup-boxes.js]: I think that with three equals you should compare against `'undefined'` instead of `undefined`

**Pull request merged**